src/service-worker.ts

```diff
@@ -115,8 +116,26 @@ export async function generateServiceWorker(options: AddServiceWorkerOptions):
   // NOTE:(usergenic) sw-precache generate() apparently replaces the
   // prefix on an already posixified version of the path on win32.
   swPrecacheConfig.stripPrefix = posixifyPath(buildRoot);
+
+  if (options.basePath) {
+    let replacePrefix = options.basePath;
+    // Avoid a "/foo//bar" situation.
+    if (replacePrefix.endsWith('/') &&
+        !swPrecacheConfig.stripPrefix.endsWith('/')) {
+      replacePrefix = replacePrefix.slice(0, -1);
+    }
+    if (swPrecacheConfig.replacePrefix) {
+      console.warn(
```

Comment:
The semantic of warn seems strange given this warning only occurs because the option is provided.  I guess its warning because this is the kind of thing that might trip people up?